File: src/Inverse Square Root Algorithms/nrisr.py
import math
import time


def newton_raphson_inv_sqrt(x, iterations):
    y = x
    for _ in range(iterations):
        y = y * (1.5 - (x * 0.5 * y * y))
    return y


x = 2.0
iterations = 5
start_time = time.time()
result = newton_raphson_inv_sqrt(x, iterations)
end_time = time.time()

print(f"Newton-Raphson Inverse Square Root of {x}: {result}")
print(f"Execution time: {(end_time - start_time) * 1e6:.2f} microseconds")

File: src/Inverse Square Root Algorithms/nrisr.c
#include <math.h>
#include <stdio.h>
#include <time.h>

float newtonRaphsonInvSqrt(float x, int iterations) {
    float y = x;
    for (int i = 0; i < iterations; i++) {
        y = y * (1.5f - (x * 0.5f * y * y));
    }
    return y;
}

int main() {
    float x = 2.0f;
    int iterations = 5;
    clock_t start, end;
    float result;

    start = clock();
    result = newtonRaphsonInvSqrt(x, iterations);
    end = clock();

    printf("Newton-Raphson Inverse Square Root of %f: %f\n", x, result);
    printf("Execution time: %f seconds\n", (double)(end - start) / CLOCKS_PER_SEC);

    return 0;
}